### PR TITLE
ExceptionlessTarget - ProcessQueueAsync for ExceptionlessClient on Flush

### PR DESCRIPTION
**Pull request by snakefoot**
Better user-experience when using NLog, and [NLog InternalLogger](https://github.com/NLog/NLog/wiki/Internal-Logging) is the recommended place when troubleshooting.

### CODE REVIEW

**Comment by snakefoot** [comment]
@niemyjski  I can see that #241 changed the default LogLevel from Trace to Warn. Would it make sense that NLog LoggingRule controlled what LogLevel should produce output?

**Comment by niemyjski** [comment]
@snakefoot yeah, I was considering that the logging targets could just set the default min log level on the client (which is only used until server settings were retrieved and last call to that method wins.

File: src/Platforms/Exceptionless.NLog/ExceptionlessTarget.cs [hunk 1]
```diff
@@ -62,5 +71,9 @@ public class ExceptionlessTarget : TargetWithLayout {
 
             builder.Submit();
         }
+
+        protected override void FlushAsync(AsyncContinuation asyncContinuation) {
+            _client.ProcessQueueAsync().ContinueWith(t => asyncContinuation(t.Exception));
```
[the remark this answers is not in the document]
Comment (snakefoot) [reply src/Platforms/Exceptionless.NLog/ExceptionlessTarget.cs:76]: NLog only calls Flush on process-exit, or if the user explict calls `NLog.LogManager.Flush()` or `NLog.LogManager.Shutdown()`

**Comment by snakefoot** [comment]
Sounds like a sensible way of handling minimum-loglevel, that you perform a "handshake" on first logevent, instead of always using LogLevel.Warn as default. Much more user-friendly.

**Pull request merged**